jwt-bearer-auth/test/Web/Auth/Bearer/JWT/Internal/CacheSpec.hs

```diff
@@ -3,43 +3,35 @@ module Web.Auth.Bearer.JWT.Internal.CacheSpec (spec) where
 import Prelude
 
 import Control.Lens
-import Control.Monad.Logger.Aeson
 import Crypto.JOSE
 import Crypto.JWT
-import qualified Data.ByteString as BS
 import Test.Hspec
 import Test.Hspec.QuickCheck
-import Web.Auth.Bearer.JWT
 import Web.Auth.Bearer.JWT.Internal.Cache
 import Web.Auth.Bearer.JWT.Test
 
-type AuthError a = JWKCacheError (BearerAuthError a)
-
 spec :: Spec
-spec = do
+spec = modifyMaxSuccess (`div` 7) $ parallel $ do
   describe "JWK cache with pure \"fetch\" action" $ do
-    prop @(TestJWKSet -> TestJWK -> Expectation) "with test JWK set"
-      $ \(TestJWKSet (JWKSet otherJWKs)) (TestJWK testJWK) -> do
-        let jwkSet = JWKSet (testJWK : otherJWKs)
-        withJWKCacheFrom (pureJWKCache jwkSet) $ \jwkCache -> do
-          Right inputJWT :: Either (AuthError JWTError) SignedJWT <-
-            makeSignedTestJWT testJWK
-          eVerifiedClaims :: Either (AuthError JWTError) ClaimsSet <-
-            runJOSELogging
-              $ verifyTokenClaims jwkCache (BS.toStrict $ encodeCompact inputJWT)
-          Right extractedClaims :: Either (AuthError JWTError) ClaimsSet <-
-            runJOSE $ unsafeGetJWTClaimsSet inputJWT
-          eVerifiedClaims ^? _Right `shouldBe` Just extractedClaims
-  describe "empty JWK cache" $ do
-    prop @(TestJWK -> Expectation) "with test JWK set"
-      $ \(TestJWK testJWK) -> do
-        withJWKCacheFrom (pureJWKCache mempty) $ \jwkCache -> do
-          Right inputJWT :: Either (JWKCacheError (AuthError JWTError)) SignedJWT <-
-            makeSignedTestJWT testJWK
-          eVerifiedClaims :: Either (AuthError JWTError) ClaimsSet <-
-            runJOSELogging
-              $ verifyTokenClaims jwkCache (BS.toStrict $ encodeCompact inputJWT)
-          eVerifiedClaims `shouldBe` Left (_JWTError . _Error # NoUsableKeys)
+    prop "with test JWK set" $ do
+      JWKSet otherJWKs <- generateJWKSet
+      testJWK <- generateJWK
+      testAudience <- generateAudience
+      let jwkSet = JWKSet (testJWK : otherJWKs)
+      withJWKCacheFrom (pureJWKCache jwkSet) $ \jwkCache -> do
+        testClaims <- generateClaimsSet testAudience
+        inputJWT <- signTestJWT testJWK testClaims
+        eVerifiedClaims <-
+          verifyTestTokenClaims jwkCache testAudience (encodeToStrict inputJWT)
+        eVerifiedClaims ^? _Right `shouldBe` Just testClaims
```

Comment:
FWIW, I find it's always better to do,

```suggestion
        eVerifiedClaims `shouldBe` Right testClaims
```

This way, when it fails you see the `Left "whatever"` in the error, instead of `Nothing`.

You do now need `Eq`/`Show` in the `e` type, which is not always possible, but worth it if you can.

Reply:
Tried to make them all be that way and just missed this one, I think, will fix